rest-api/alembic/versions/04564bf1b5e8_deidentify_ppi_views.py

```diff
@@ -0,0 +1,279 @@
+"""Deidentify ppi views
+
+Revision ID: 04564bf1b5e8
+Revises: 6f9266e7a5fb
+Create Date: 2017-12-20 13:21:37.487290
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, SuspensionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = '04564bf1b5e8'
+down_revision = '6f9266e7a5fb'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+_QUESTIONNAIRE_RESPONSE_ANSWER_VIEW_SQL = """
+CREATE OR REPLACE VIEW questionnaire_response_answer_view AS
+ SELECT
+   p.participant_id participant_id,
+   YEAR(p.sign_up_time) participant_sign_up_year,
+   p.withdrawal_status participant_withdrawal_status,
+   YEAR(p.withdrawal_time) participant_withdrawal_year,
+   p.suspension_status participant_suspension_status,
+   YEAR(p.suspension_time) participant_suspension_year,
+   hpo.name hpo,
+   (SELECT GROUP_CONCAT(concept_code.value SEPARATOR ',') FROM questionnaire_concept qc
+     INNER JOIN code concept_code ON concept_code.code_id = qc.code_id
+     WHERE qc.questionnaire_id = q.questionnaire_id) module,
+   q.questionnaire_id questionnaire_id,
+   q.created questionnaire_creation_time,
+   qr.questionnaire_response_id questionnaire_response_id,
+   YEAR(qr.created) questionnaire_response_submission_year,
+   qc.value question_code,
+   qc.code_id question_code_id,
+   YEAR(qra.end_time) answer_end_year,
+   ac.value answer_code,
+   ac.code_id answer_code_id,
+   qra.value_boolean answer_boolean,
+   qra.value_decimal answer_decimal,
+   qra.value_integer answer_integer,
+   (qra.value_string != '') answer_string_present,
```

Comment:
From this:

https://stackoverflow.com/questions/3059805/difference-between-mysql-is-not-null-and

It sounds like checking != '' will result in NULL when value_string is NULL, and false when it's ''. 

I think in our data we normally have NULL, rather than '' for strings.

In any case, it's a bit weird to treat empty string and NULL differently here. Perhaps we should use COALESCE or something to avoid this? Definitely let's test the null cases...

Reply:
So I think from an analysis standpoint, it's probably(?) better to have always a 0 or 1 for these "present" fields... easier to deal with than three possible values.

Reply:
Maybe, though it makes this inconsistent from the other columns here. FTR there were only two possible output values here, NULL and 1.

I'm not particularly happy with any of the above solutions though, so I updated to 0/1.